src/charm.py

```diff
@@ -116,15 +166,18 @@ def _configure_workload(self, event: ops.charm.EventBase) -> None:
         self.ingress.update_config({"service-hostname": self.config["external_hostname"]})
 
         container = self.unit.get_container(CONTAINER_NAME)
+        statsd_container = self.unit.get_container("statsd-prometheus-exporter")
```

Comment:
I think we should remove `CONTAINER_NAME` as a variable and just refer to "gunicorn" when we mean the gunicorn container and "statsd-prometheus-exporter" when we mean this container as you do here. It's confusing having one variable that's `CONTAINER_NAME`  when we have two different containers (and I don't think it's worth opening and parsing a yaml file on every charm invocation - additionally because we're basing it of the charm name currently this will break if someone tries to create a personal branch of this charm such as mthaddon-gunicorn-k8s).